pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/invertedindex/RealtimeInvertedIndexReader.java

```diff
@@ -60,6 +60,13 @@ public MutableRoaringBitmap getDocIds(int dictId) {
     ThreadSafeMutableRoaringBitmap bitmap;
     try {
       _readLock.lock();
+      // NOTE: the given dict Id might not be added to the inverted index yet. We first add the value to the dictionary.
+      // Before the value is added to the inverted index, the query might have a predicate which matches the newly added
+      // value. In that case, the given dictionary Id does not exist in the inverted index, and we return an empty
+      // bitmap.
+      if (_bitmaps.size() == dictId) {
```

Comment:
I don’t understand this check. This is still prone to error rt. Imagine two new dict ids added

Also just thinking out loud. the previous behavior was probably better than returning empty bitmap?  What happens in Not In case.

Reply:
Previous behavior caused `IndexOutOfBounds` error when the race condition happened (the query hits when the value got added to dictionary but not yet in the inverted index). I thought that it's only possible to meet the case where `bitmap.size() == dictId` since the consumption is happening in a single thread. @Jackie-Jiang can explain it in more detail and also comment on `not in` case.

Reply:
@kishoreg Actually you are right for multi-valued column. I'll change this to `_bitmaps.size() <= dictId` to handle all cases.
In the ancient code, we might return null, but then we refactored the code and uniformed the interface. Right now the interface does not expect null return value, and this is corner case which happens rarely, so I would rather handle this inside the implementation by returning an empty bitmap instead of null.

Reply:
cool.

Another thought I had was, should we even check for the _bitmaps.size() <= dictId. What if we return empty one if the bitmap is null.

Reply:
Ignore the last comment. I forgot to refresh the PR and was looking at old code.